wal/wal.go

```diff
@@ -131,22 +132,41 @@ func Create(dirpath string, metadata []byte) (*WAL, error) {
 		return nil, err
 	}
 
-	// rename of directory with locked files doesn't work on windows; close
-	// the WAL to release the locks so the directory can be renamed
-	w.Close()
-	if err := os.Rename(tmpdirpath, dirpath); err != nil {
-		return nil, err
+	if runtime.GOOS == "windows" {
```

Comment:
Split these separate code paths into the files `wal_windows.go` and `wal_unix.go` with `windows` and `!windows` build constraints? Each file would provide a platform specific implementation of `func (w *WAL) renameWal(tmpdirpath string) (*WAL, error)` for `Create` to call.